core/launcher/src/manager/app_launcher.rs

```diff
@@ -513,8 +515,43 @@ impl AppLauncher {
         }
     }
 
-    fn get_transport(url: String) -> AppRuntimeTransport {
-        if !url.as_str().contains("__firebolt_endpoint") {
+    fn get_modified_url(manifest: AppManifest, sessionid: &str) -> String {
+        let url = manifest.start_page.to_string();
```

Comment:
I think this is not required, anyway, we have to pass a reference to Url::parse. so we can call it as Url::parse(&manifest.start_page)

Reply:
changes added.